pkgs/build-support/rust/rustc-wrapper/default.nix

```diff
@@ -46,24 +52,50 @@ runCommand "${rustc-unwrapped.pname}-wrapper-${rustc-unwrapped.version}"
         targetPlatforms
         badTargetPlatforms
         ;
-      unwrapped = rustc-unwrapped;
+      unwrapped = package;
     };
 
-    meta = rustc-unwrapped.meta // {
-      description = "${rustc-unwrapped.meta.description} (wrapper script)";
+    meta = package.meta // {
+      description = "${package.meta.description} (wrapper script)";
       priority = 10;
     };
   }
-  ''
-    mkdir -p $out/bin
-    ln -s ${rustc-unwrapped}/bin/* $out/bin
-    rm $out/bin/{rustc,rustdoc}
-    prog=${rustc-unwrapped}/bin/rustc extraFlagsVar=NIX_RUSTFLAGS \
-        substituteAll ${./rustc-wrapper.sh} $out/bin/rustc
-    prog=${rustc-unwrapped}/bin/rustdoc extraFlagsVar=NIX_RUSTDOCFLAGS \
-        substituteAll ${./rustc-wrapper.sh} $out/bin/rustdoc
-    chmod +x $out/bin/{rustc,rustdoc}
-    ${lib.concatMapStrings (output: "ln -s ${rustc-unwrapped.${output}} \$${output}\n") (
-      lib.remove "out" rustc-unwrapped.outputs
-    )}
-  ''
+  (
+    if flavor == "rustc" then
+      ''
+        mkdir -p $out/bin
+        ln -s ${package}/bin/* $out/bin
+        rm $out/bin/{rustc,rustdoc}
+        prog=${package}/bin/rustc extraFlagsVar=NIX_RUSTFLAGS \
+            substituteAll ${./rustc-wrapper.sh} $out/bin/rustc
+        prog=${package}/bin/rustdoc extraFlagsVar=NIX_RUSTDOCFLAGS \
+            substituteAll ${./rustc-wrapper.sh} $out/bin/rustdoc
+        chmod +x $out/bin/{rustc,rustdoc}
+        ${lib.concatMapStrings (output: "ln -s ${package.${output}} \$${output}\n") (
+          lib.remove "out" package.outputs
+        )}
+      ''
+    else if flavor == "clippy" then
+      ''
+        mkdir -p $out/bin
+        prog=${package}/bin/clippy-driver extraFlagsVar=NIX_CLIPPY_DRIVERFLAGS \
+            substituteAll ${./clippy-wrapper.sh} $out/bin/clippy-driver
```

Comment:
Does it need a different wrapper?

Reply:
It needs the sysroot argument after the user provided arguments, not before and I wasn't sure what would be the best way to do that with the existing rustc wrapper.

Reply:
Can't we just set `prog="${package}/bin/clippy-driver rustc"`?

Reply:
Okay, so what if we substitute a number into the wrapper, for the number of arguments to shift from `$@` and move into some other array that we expand after `prog`?

Reply:
From looking at the clippy-driver code it looks like it can also be invoked without the rustc argument, so I'm not sure about all of this. It looks like it can also take a SYSROOT env var: https://github.com/rust-lang/rust-clippy/blob/034136ddfaee774cecb6c36c0f33f499bf7bd149/src/driver.rs#L260

Reply:
Hmm, okay. In that case maybe it's better to skip the wrapper and use the environment variable. It won't be quite right in cases where the wrapper does other things, but still better than now?